content/sips/sip-230.md

```diff
@@ -0,0 +1,99 @@
+
+Implementation of SIP-165, and to some extent, futures, have revealed that we need a more self-contained, generic system for chainlink oracle validation is required. To solve this problem,
+a new circuit breaker contract which all queries to chainlink aggregators must run through is created.
+
+For testing purposes, this class can be used to reduce the provisioning complexity of the system in general by being able to set override prices from the
+circuit breaker system itself.
+
+## Motivation
+
+<!--This is where you explain the reasoning behind how you propose to solve the problem. Why did you propose to implement the change in this way, what were the considerations and trade-offs? The rationale fleshes out what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+This is a technical debt item which will improve the maintainability and security of the protocol
+
+### Technical Specification
+
+#### `OracleManager`
+
+This is a new contract added to the system which will replace `ExchangeCircuitBreaker`, portions of `ExchangeRates` where aggregators are stored,
+and any other circuit breaker functionality which previously existed will be removed and replaced with a call to this contract.
+
+This contract will include the following external methods:
+
+* `addAggregator(bytes32 id, address target)`: register chainlink compatible oracle from the given target, referenced with the given `id`. Similar to existing `ExchangeRates.addAggregator`
+* `removeAggregator(bytes32 id, address target)`: opposite of `addAggregator`
+* `reset(bytes32 id, uint value)`: If a circuit breaker was triggered, disarm it and initialize it to the given value for last known
+* `getCurrentRound(bytes32 id)`: Returns the current round ID reported by the oracle
+* `getValue(bytes32 id, uint round) returns (uint value)`: Retrieve a value from the oracle referenced by `id`. Before returning, the value will be passed through a circuit breaker function (defined below). If the value is/has tripped the breaker, or is out of date, the function will fail internally
+* `failureTimestamp(bytes32 id)`: used to get the time at which a breaker was tripped or when the last update was (if the value is stale)
+* `getValues(bytes32 id, uint roundStart, uint roundEnd) returns (uint[] values, uint broken)`: Same as `getValue,` but retrieves a range of values between `roundStart` and `roundEnd`
+* `setValue(bytes32 id, uint value)`: Allows owner to manually set an aggregator price to a static value. Useful for test overrides or defined prices like `sUSD`. Not compatible with an aggregator previously added using `addAggregator`.
+
+#### Circuit Breaker Functionality
```

Comment:
the second return parameter `broken`  is designed to handle this in `getValue` or its variants. Basically, if its broken, the calling code can decide what to do. We actually follow this pattern in most places within the synthetix system already, with the result most ofent just being passed up the line if its a `view` function, or the execution stopped without reverting if its a `mutate` function

Reply:
Mutative vs. non-mutative methods should be considered. This is because this contract's method will called by both mutative methods (which should than set the cached values in this contract) which would return a `broken` flag, and views - which would only return a `invalid` flag.